### PR TITLE
Make HttpMethodConstraint case insensitive

### LINKED ISSUE #3551: HttpMethod case sensitivity
`HttpMethodConstraint` is case sensitive when comparing HTTP methods. Also ~~`HttpContext.Request.Method`~~ `Microsoft.AspNet.Http.Internal.DefaultHttpRequest` case is same as what passed by the client which could cause inconsistency in some cases.

*I am not sure, but I guess http.sys was changing it to upper case, but Kestrel preserve.*

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #3551

### CODE REVIEW

File: test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections.Generic;
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs]: License header

File: test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+using Microsoft.AspNet.Http.Internal;
+using Microsoft.AspNet.Mvc.Abstractions;
+using Microsoft.AspNet.Mvc.ActionConstraints;
+using Microsoft.AspNet.Routing;
+using Microsoft.Extensions.Primitives;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.Core.Test
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs]: :-1:

File: test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+{
+    public class HttpMethodConstraintTest
+    {
+        [Theory]
+        [InlineData("gEt")]
+        [InlineData("get;GeT")]
+        [InlineData("get")]
+        [InlineData("GET")]
+        public void HttpMethodConstraint_Accept_Preflight_CaseInsensitive(string httpMethodStr)
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs]: `httpMethodStr` horrible

File: test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        {
+            // Arrange
+            var httpMethods = httpMethodStr.Split(';');
+            var constraint = new HttpMethodConstraint(httpMethods);
+
+            var context = new ActionConstraintContext();
+            var routeContext = CreateRouteContext("oPtIoNs", true);
+            
+            var action = new ActionDescriptor();
+            var constraints = new List<IActionConstraint> {
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs]: This isn't java

File: test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        private static RouteContext CreateRouteContext(string requestedMethod, bool preflight = false)
+        {
+            var httpContext = new DefaultHttpContext();
+
+            httpContext.Request.Method = requestedMethod;
+
+            if(preflight)
+            {
+                httpContext.Request.Headers.Add("Origin", StringValues.Empty);
+                httpContext.Request.Headers.Add("Access-Control-Request-Method", "GET");
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs]: I'd expect to see this case vary - it just kinda throws me when looking at the tests.

File: test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+            Assert.True(result);
+        }
+
+        private static RouteContext CreateRouteContext(string requestedMethod, bool preflight = false)
+        {
+            var httpContext = new DefaultHttpContext();
+
+            httpContext.Request.Method = requestedMethod;
+
+            if(preflight)
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs]: Fix spacing. You do this too much. Get used to `CTRL+K, D`

File: test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs (hunk 4 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/HttpMethodConstraintTest.cs]: https://www.youtube.com/watch?v=_W25lPno3FE

**Comment by ryanbrandenburg** [comment]
:up: :date:

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**